### DISCUSSION TITLE
Conditional saving of cache

### DISCUSSION THREAD

**Discussion by maru-sama** [JanDeDobbeleer/oh-my-posh#1681] [Ideas]
Good Morning Jan,

Quick question regarding the cache. If I am not mistaken the cache file is written every time oh-my-posh runs regardless if anything has changed or not. While negligible on "normal" machines it is a different story for my Pis running on sd-cards. I am not sure about an increase of speed since the file cache should handle these writes but I am more worried about the wear on the sd-card over a long time. Would it be possible to only write the cache file if there were actual changes to the values?

**Comment by JanDeDobbeleer** [comment]
That's definitely possible to implement, yes.

**Reply by maru-sama** [reply]
I did something like this locally as a quick proof of concept, but maybe there is a better way

```--- a/src/environment/cache.go
+++ b/src/environment/cache.go
@@ -44,8 +44,12 @@ func (fc *fileCache) Close() {
        if len(cache) == 0 {
                return
        }
-       if dump, err := json.MarshalIndent(cache, "", "    "); err == nil {
-               _ = ioutil.WriteFile(fc.cachePath+fileName, dump, 0644)
+       _, found := cache["dirty"]
+       if found {
+               delete(cache, "dirty")
+               if dump, err := json.MarshalIndent(cache, "", "    "); err == nil {
+                       _ = ioutil.WriteFile(fc.cachePath+fileName, dump, 0644)
+               }
        }
 }

@@ -78,4 +82,9 @@ func (fc *fileCache) Set(key, value string, ttl int) {
                Timestamp: time.Now().Unix(),
                TTL:       ttl,
        })
+       fc.cache.set("dirty", &cacheObject{
+               Value:     "1",
+               Timestamp: time.Now().Unix(),
+               TTL:       -1,
+       })
 }
```

**Reply by JanDeDobbeleer** [reply]
Will be live today: https://github.com/JanDeDobbeleer/oh-my-posh/pull/1712/commits/64e83851adbca70af954bb32154f7113ef94d654

**Reply by maru-sama** [reply]
Perfect,

I was just working on it too and had a very similar solution. The one above was way too complicated :) Thank you very much for taking care of of this.
